servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -1429,6 +1429,60 @@ Size2 TextureStorage::texture_size_with_proxy(RID p_proxy) {
 	return texture_2d_get_size(p_proxy);
 }
 
+void TextureStorage::texture_rd_initialize(RID p_texture, const RID &p_rd_texture) {
```

Comment:
I wonder, should this RID be also freed when the texture is freed, or should it be freed manually from RenderingDevice?
What do you think?

Reply:
I'm a strong believe over the design pattern that the entity that creates a resource should also free it. So if I create the texture on RenderingDevice I should also free it.

One thing that we could do here, and that solves some other problems too, is that this always creates a new shared texture on the texture given to this function. Then destroying this texture destroys the texture view, but you're still left with the original texture to destroy on the rendering device.

Reply:
Another question around this that could make things difficult for us. In theory someone could create a texture object through the `RenderingServer`, call `texture_get_rd_texture` to get the RID to the RD texture, and then call `texture_rd_create` to create a new texture object for that RD texture.
There is however no dependency between the original texture and the new texture meaning that if the original texture is freed, our new texture object will contain an invalid RID.

Reply:
Come to think of it, we should expose `texture_get_rd_texture` to users as well. So far this has only been used internally for XR I think.

Reply:
I guess we can just throw an error in ``texture_rd_create()`` if it is called using an RD texture that already belongs to a RS::Texture. But I guess that only works if we track dependencies on the RD side, the way we do on the RS side

Reply:
Yeah we currently don't track those dependencies and I feel we already have a big memory impact with the way we track those atm.

Reply:
I guess this is fixed by using a shared texture instead of creating a new texture right?

Reply:
Yeah I think it works this way now